graphql-ballerina/modules/parser/parser.bal

```diff
@@ -164,13 +177,32 @@ public class Parser {
     }
 
     isolated function addFragmentToNode(ParentNode parentNode) returns ([string, Location]|Error) {
-        Token token = check self.readNextNonSeparatorToken(); // Consume Ellipsis token
-        token = check self.readNextNonSeparatorToken();
+        Token token = check self.readNextNonSeparatorToken();
         string fragmentName = check getIdentifierTokenvalue(token);
         parentNode.addFragment(fragmentName);
         return [fragmentName, token.location];
     }
 
+    isolated function addInlineFragmentToNode(ParentNode parentNode) returns ([string, Location]|Error) {
+        Token token = check self.readNextNonSeparatorToken();//Consume on keyword
+        Location location = token.location.clone();
```

Comment:
Is this location correct? The fragment location should be the location where we create the fragment, which is the starting column of the ELLIPSIS token. We have to come up with a way to preserve the location of each inline fragment. Consider the following query:

```
{
  __schema {
    types {
      ... on Student {
        name
      }
      ... on Student {
        name
      }
    }
  }
}
```

This will send an error response:

```json
{
  "errors": [
    {
      "message": "Fragment cannot be spread here as objects of type \"__Type\" can never be of type \"Student\".",
      "locations": [
        {
          "line": 4,
          "column": 7
        }
      ]
    },
    {
      "message": "Fragment cannot be spread here as objects of type \"__Type\" can never be of type \"Student\".",
      "locations": [
        {
          "line": 7,
          "column": 7
        }
      ]
    }
```
Check the column numbers of the error response.
Let's add a test to check this functionality.

Reply:
Tracked with [#1331](https://github.com/ballerina-platform/ballerina-standard-library/issues/1331)